Deploy-CAM.ps1

```diff
@@ -905,54 +982,48 @@ function New-PopulatedKeyvault()
+			Remove-Item $certificateFile
+		}
 
-		#export pfx file
-		Export-PfxCertificate -Cert $certPath -FilePath $certPfx -Password $secureCertPswd
+		#generate password for pfx file
+        #https://docs.microsoft.com/en-us/azure/application-gateway/application-gateway-ssl
+		#The certificate password must be between 4 to 12 characters made up of letters or numbers. Special characters are not accepted.
+		$certPswd = -join ((48..57) + (65..90) + (97..122) | Get-Random -Count 10 | % {[char]$_})
 
-		Write-Host "Putting certificate in Key Vault."
+		$certificateFilePassword = ConvertTo-SecureString -String $certPswd -AsPlainText -Force
 
-		#read from pfx file and convert to base64 string
-		$fileContentEncoded = [System.Convert]::ToBase64String([System.IO.File]::ReadAllBytes($certPfx))
+		#export pfx file
+		Export-PfxCertificate -Cert $certPath -FilePath $certificateFile -Password $certificateFilePassword
 
-		$CSCertificate = ConvertTo-SecureString $fileContentEncoded -AsPlainText -Force
+        #delete self-signed certificate
+		if(Test-Path $certPath) { 
+			Remove-Item $certPath -ErrorAction SilentlyContinue
+		}
+	} 
 
-		$CSCertSecretName = $CAMConfig.internal.CSCertSecretName
-		$CSCertSecret = Set-AzureKeyVaultSecret -VaultName $kvName -Name $CSCertSecretName -SecretValue $CSCertificate
-		$CSCertSecretVersionedURL = $CSCertSecret.Id
-		$CSCertSecretURL = $CSCertSecretVersionedURL.Substring(0, $CSCertSecretVersionedURL.lastIndexOf('/'))
+	#read from pfx file and convert to base64 string
+	$fileContentEncoded = [System.Convert]::ToBase64String([System.IO.File]::ReadAllBytes($certificateFile))
 
-		$CSCertPasswordSecretName = $CAMConfig.internal.CSCertPasswordSecretName
-		$CSCertPasswordSecret = Set-AzureKeyVaultSecret -VaultName $kvName -Name $CSCertPasswordSecretName -SecretValue $secureCertPswd
-		$CSCertPasswordSecretVersionedURL = $CSCertPasswordSecret.Id
-		$CSCertPasswordSecretURL = $CSCertPasswordSecretVersionedURL.Substring(0, $CSCertPasswordSecretVersionedURL.lastIndexOf('/'))
+	$CSCertificate = ConvertTo-SecureString $fileContentEncoded -AsPlainText -Force
 
-		Write-Host "Successfully put certificate in Key Vault."
-		
-		$secretHash = @{}
-		$secretHash.Add($rcSecretName,$rcSecretURL)
-		$secretHash.Add($djSecretName,$djSecretURL)
-		$secretHash.Add($rwLocalSecretName,$rwLaSecretURL)
-		$secretHash.Add($csLocalSecretName,$csLaSecretURL)
-		$secretHash.Add($CSCertSecretName,$CSCertSecretURL)
-		$secretHash.Add($CSCertPasswordSecretName,$CSCertPasswordSecretURL)
+	$certInfo = @{
+		"cert" = $CSCertificate;
+		"passwd" = $certificateFilePassword
 	}
-	catch {
-		throw
-	}
-	finally {
-		#done with files
-		if(Test-Path $certPfx) { Remove-Item $certPfx -ErrorAction SilentlyContinue }
-		if(Test-Path $certPath) { Remove-Item $certPath -ErrorAction SilentlyContinue }
-	}
-	return $secretHash
-}
 
+	#delete certificate file if it is generated
+	if ($needToCreateSelfCert -and	(Test-Path  $certificateFile) ) { 
```

Comment:
This used to be in a 'finally' block to ensure it's always (or rather - more often) - properly cleaned up. Does that make sense in this case?

Reply:
To remove the duplicated the code to read certificate file in both cases which are generating on the fly or providing certificate file. the deleting code is outside of try block and it only delete the generating certificate file.